### DISCUSSION TITLE
Multi cursor doesn't work when screen scroll

### DISCUSSION THREAD

**Discussion by skuong** [vscode-neovim/vscode-neovim#2198] [Q&A] [answered]
In the [vscode.with_insert](https://github.com/vscode-neovim/vscode-neovim/tree/master?tab=readme-ov-file#vscodewith_insertcallback) section, we have a code example:
```lua
vim.keymap.set({ "n", "x", "i" }, "<C-d>", function()
  vscode.with_insert(function()
    vscode.action("editor.action.addSelectionToNextFindMatch")
  end)
end)
```
To make it work, I need to remove `d` from "Ctrl Keys For Insert Mode" and "Ctrl Keys For Normal Mode".

Issue: When there are a lot of matches that cause the screen to scroll, the selection just go back to the first one.
I already set "vscode-neovim.neovimClean": true.

Anyone have the same problem?

**Comment by xiyaowong** [comment]
Please provide a screen recording.

**Comment by skuong** [comment]
https://github.com/user-attachments/assets/fb576758-bf7a-43b7-a885-0379ca0ad8ec

It happens when I use mouse to scroll as well.

**Reply by xiyaowong** [reply, marked as answer]
You didn't send `<C-d>` to nvim.

https://github.com/vscode-neovim/vscode-neovim#keybinding-passthroughs

**Reply by skuong** [reply]
I understand now. Thank you so much.

**Comment by xiyaowong** [comment]
> To make it work, I need to remove `d` from "Ctrl Keys For Insert Mode" and "Ctrl Keys For Normal Mode".

Completely the opposite! You should add `d` to these options.

**Reply by skuong** [reply]
When I send `d` to nvim, `ctrl+d` would scroll page down instead. Is there other vscode settings I need to do?

**Reply by xiyaowong** [reply]
Did you add this to your config?
```lua
vim.keymap.set({ "n", "x", "i" }, "<C-d>", function()
  vscode.with_insert(function()
    vscode.action("editor.action.addSelectionToNextFindMatch")
  end)
end)
```

**Reply by skuong** [reply]
Yes I did.

It works now.
I didn't check the link in your previous reply (the one that I marked as answer). My bad.

Thank you so much, xiyaowong. You're really helpful.
